src/renderers/WebGLRenderer.js

```diff
@@ -2476,7 +2476,7 @@ THREE.WebGLRenderer = function ( parameters ) {
 
 		var updateBuffers = false,
 			wireframeBit = material.wireframe ? 1 : 0,
-			geometryHash = ( geometry.id * 0xffffff ) + ( program.id * 2 ) + wireframeBit;
+			geometryHash = ( geometry.id * 0x1000000 ) + ( program.id * 4 ) + 2 + wireframeBit;
```

Comment:
Maybe it would be easier if it was just this?

```javascript
geometryHash = 'buffer' + geometry.id + '_' + program.id + '_' + wireframeBit;
```

And

```javascript
geometryHash = geometryGroup.id + '_' + program.id + '_' + wireframeBit;
```